core/lib/network-request.js

```diff
@@ -124,11 +124,25 @@ class NetworkRequest {
     this.parsedURL = /** @type {ParsedURL} */ ({scheme: ''});
     this.documentURL = '';
 
-    this.startTime = -1;
-    /** @type {number} */
-    this.endTime = -1;
-    /** @type {number} */
-    this.responseReceivedTime = -1;
+    /** ms. When the main thread initially discovers a network request. */
+    this.mainThreadStartTime = -1;
+    /**
+     * ms. When the network service is about to handle a request, ie. just before going to the
+     * HTTP cache or going to the network for DNS/connection setup.
+     */
+    this.networkRequestTime = -1;
```

Comment:
for symmetry's sake, maybe this should be `networkStartTime`?

With `networkEndTime` it makes a nice counterpart to `mainThreadStartTime`/`mainThreadEndTime`, making it more obvious which properties mark start/end of durations without requiring much thought.

Reply:
Am OK with this, @paulirish ?

Reply:
Hmmmmmmmmmmm. I acknowledge that 'start' is more consistent.. but...........

semantically its not a "start"

<img width="415" alt="image" src="https://user-images.githubusercontent.com/39191/198557631-41de2647-3004-4954-b4b0-4591200a9e66.png">

the 'requestTime' is the line between 'queueing' and 'stalled'. but you can see that 'request sent' can be quite a while after it.  so... yah.

and also.. of all the times in https://nettimings.surge.sh/Network-Timings.svg "requestTime" is like the only one that has just 1 name everywhere. Good thing too since it's definition is so weird.